Adding a temporary fix for kubectl get output e2e test

robscott opened the pull request:
**What type of PR is this?**
/kind failing-test

**What this PR does / why we need it**:
When running older e2e tests against a newer Kubernetes release, any new API resources would result in errors due to a lack of test data in the older tests. This temporarily disables the test for the new IngressClass resource until we can find a better approach here. This will only fix the root issue for the skew tests if this fix is also cherry picked into 1.17.

**Which issue(s) this PR fixes**:
Fixes #88335

**Does this PR introduce a user-facing change?**:
```release-note
NONE
```

/sig cli
/priority critical-urgent
/cc @soltysh 
/assign @alejandrox1 @BenTheElder

alejandrox1 reviewed (commented):
What if we commit this directly to 1.17?
That should, hypothetically, fix the kubectl skew test and keep the valid e2e configuration for 1.18 onwards, no?

robscott commented:
@alejandrox1 That's a good idea, didn't even realize that was possible. As I'm thinking about it more, I wonder if there's a better more long term approach here, to have the skew test suite avoid tests that are fundamentally incompatible with it, maybe with some kind of tag? Not sure what the best approach is, but that would certainly be more future proof than this.

alejandrox1 commented:
> @alejandrox1 That's a good idea, didn't even realize that was possible. As I'm thinking about it more, I wonder if there's a better more long term approach here, to have the skew test suite avoid tests that are fundamentally incompatible with it, maybe with some kind of tag? Not sure what the best approach is, but that would certainly be more future proof than this.

did some digging here https://github.com/kubernetes/kubernetes/issues/88335#issuecomment-595578620

A new tag may be an interesting take,specially long term. A bit tricky since I think we may run into similar issues and retroactively add the tag when a test is not meant to pass as is with a new/older version ? (just thinking out loud here)

liggitt commented:
what is the test failure?

robscott commented:
@liggitt I'm very open to other approaches here. This never felt like an ideal approach, but it did work. It seems like there are a number of ways to potentially mess up kubectl integration, and this is an e2e way to cover at least the headers of output. When this test was originally created (https://github.com/kubernetes/kubernetes/pull/83748) I think we all wanted to have a simpler kubectl setup that would also enable simpler testing, but this was the least bad available option we could come up with at the time.

droslean commented:
/retest

BenTheElder commented:
I don't think I have enough context on skew testing but I want to unblock fixing our tests...
/approve

BenTheElder commented:
approving purely as a reasonable-ish test-only change, I'm on the fence about the approach based on above discussion but it's not clear that we have an alternative ready.

liggitt commented:
if this test is not failing in master, we should not exclude this type in master. "fixing" it in 1.17 (and maybe 1.18?) is reasonable as a stop-gap. Opened https://github.com/kubernetes/kubernetes/pull/88966 as a more systemic fix to make sure all resources define table converters.

liggitt commented:
/lgtm
/approve

as a temporary fix for 1.17/1.18

liggitt commented:
/retest

alejandrox1 commented:
/test pull-kubernetes-e2e-gce
prev err: https://prow.k8s.io/view/gcs/kubernetes-jenkins/pr-logs/pull/88883/pull-kubernetes-e2e-gce/1237026949231022081/

liggitt commented:
/hold

actually... can we just make this change against 1.17? there's no real reason to exempt it from the 1.18 test

alejandrox1 commented:
> /hold
> 
> actually... can we just make this change against 1.17? there's no real reason to exempt it from the 1.18 test

I would be in favor of that!

robscott committed:
Adding a temporary fix for kubectl get output e2e test

When running older e2e tests against a newer Kubernetes release, any new
API resources would result in errors due to a lack of test data in the
older tests.

robscott commented:
@liggitt Thanks for the much better solution in #88966! I've updated this PR to target 1.17 only now.

liggitt commented:
/skip
/retest

liggitt commented:
/skip

liggitt commented:
I think the rebase to 1.17 has confused github/ci. You might want to close this and reopen a new PR against 1.17

robscott commented:
👍 will do, thanks!

robscott commented:
/close
